extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/interceptor/TransactionalInterceptorBase.java

```diff
@@ -227,10 +252,18 @@ protected Object invokeInCallerTx(InvocationContext ic, Transaction tx) throws E
     }
 
     protected Object invokeInNoTx(InvocationContext ic) throws Exception {
-
+        checkConfiguration(ic);
         return ic.proceed();
     }
 
+    private void checkConfiguration(InvocationContext ic) {
+        TransactionConfiguration configAnnotation = getTransactionConfiguration(ic);
+        if (configAnnotation != null && configAnnotation.timeout() != TransactionConfiguration.UNSET_TIMEOUT) {
+            throw new RuntimeException("Changing timeout via @TransactionConfiguration can only be done " +
```

Comment:
Maybe `IllegalStateException` would be better?

Reply:
`RuntimeException` are already used in multiple area of the code so I use it, but I'm OK to switch to `IllegalStateException`. It's a developement error (a configuration error), I don't know what is the best practice in Quarkus to specify this kind of error.